### DISCUSSION TITLE
Some questions about Materials and Executioner

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#22708] [Q&A General] [answered]
Hi, experts. I have some questions.

1. In a transient simulation, can I substitute the material properties used at a certain time? My material properties are obtained from a txt file. For example, at 10s, I want to replace the material properties with another txt file, can MOOSE do this?

2. There are several physics involved in my transient simulation, let's call them A, B, C. Different physics can achieve different convergence tolerances. I mean, nl_rel_tol = 1e-4 might be small enough for A, but nl_rel_tol = 1e-10 for B physics to meet our requirements. However A, B, C are all solved in one input file. So I wonder if MOOSE can set different convergence criteria for different variable groups.

3. Does MOOSE judge as `Solving Converged` as long as one of the convergence criteria is met? For example, if I set `nl_abs_tol=1e-8` and `nl_rel_tol=1e-8` in Executioner, as long as the `nonlinear iteration residual` meets one of these two criteria, will it be judged as `Solving Converged`?

**Comment by GiudGiud** [comment, marked as answer]
Hello

1)
There's a few ways to do this.
One way is to use the Control system to substitute one material by another. You disable the old one, enable the new one using one to the ConditionalEnableControl objects
Another way is to write a checkpoint and reload from that checkpoint in another input file with the updated material

2)
We can do this using manual scaling. Scale each variables corresponding to each physics using the scaling_factor parameter in the variables block.
The residual is then scaled by this factor. see this porous flow example
https://mooseframework.inl.gov/modules/porous_flow/convergence.html

3)
Yes. As soon as you meet one of the criteria we consider the simulation converged.

Guillaume

**Reply by js-jixu** [reply]
Thank you, Guillaume. Your suggestions are very helpful.
